### PR TITLE
numbers_strings

### PR DESCRIPTION
**Pull request by avagyan98**

### CODE REVIEW

File: strings.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+// 1. Uppercase the first character
+
+function ucFirst(str) {
+    str = str[0].toUpperCase() + str.slice(1);
```
Comment (robhov97) [inline strings.js]: Also, make all the other letters lowercase. For example:
`ucFirst('sOmEtHiNg'); // Something`

File: strings.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+function checkSpam(str, checkStr1, checkStr2) {
+    str = str.toLowerCase();
+    checkStr1 = checkStr1.toLowerCase();
+    checkStr2 = checkStr2.toLowerCase();
+
+    if(str.includes(checkStr1) || str.includes(checkStr2)){
+        return true
+    }
+
+    return false
```
Comment (robhov97) [inline strings.js]: Simplify, make it on a one line

File: strings.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+let text = 'text';
+console.log(ucFirst(text));
+
+// 2. Check for spam
+
+function checkSpam(str, checkStr1, checkStr2) {
+    str = str.toLowerCase();
+    checkStr1 = checkStr1.toLowerCase();
+    checkStr2 = checkStr2.toLowerCase();
```
Comment (robhov97) [inline strings.js]: don't change the paramteres

File: strings.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    str = str[0].toUpperCase() + str.slice(1);
+    return str
+}
+
+let text = 'text';
+console.log(ucFirst(text));
+
+// 2. Check for spam
+
+function checkSpam(str, checkStr1, checkStr2) {
```
Comment (robhov97) [inline strings.js]: it should have one argument, the string that you should check

File: strings.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+    return false
+}
+
+console.log(checkSpam("asdsd sdf viagra", 'viaGra', 'XXX'));
+
+// 3. Truncate the text
+
+function truncate(str, lengthStr) {
+    return (str.length > lengthStr) ? str.slice(0, lengthStr - 3) + "..." : str
```
Comment (robhov97) [inline strings.js]: use template literals instead of `+ "..."`

File: strings.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+console.log(truncate("Hi everyone!", 10))
+
+// 4. Extract the money
+
+function extractCurrencyValue(str) {
+    let firstNumber;
+    for(let i = 0; i < str.length; i++){
+        if (str.codePointAt(i) >= 48 && str.codePointAt(i) <= 57) {
+            firstNumber = i;
+            break
```
Comment (robhov97) [inline strings.js]: you could just return the value here instead of breaking it and then returning it

File: strings.js [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+        }
+    }
+
+    return Number(str.slice(firstNumber))
+}
+
+const many = "$$340"
```
Comment (robhov97) [inline strings.js]: Just assume the case when there's only one currency sign in the beginning, no need to overengineer.

**Review by robhov97 — CHANGES_REQUESTED** [review]

File: strings.js (hunk 1 above)
Comment (avagyan98) [reply strings.js]: done

File: strings.js (hunk 2 above)
Comment (avagyan98) [reply strings.js]: done

File: strings.js (hunk 3 above)
Comment (avagyan98) [reply strings.js]: done

File: strings.js (hunk 4 above)
Comment (avagyan98) [reply strings.js]: done

File: strings.js (hunk 5 above)
Comment (avagyan98) [reply strings.js]: done

File: strings.js (hunk 6 above)
Comment (avagyan98) [reply strings.js]: done

File: strings.js (hunk 7 above)
Comment (avagyan98) [reply strings.js]: done

File: strings.js [hunk 8]
```diff
@@ -0,0 +1,37 @@
+// 1. Uppercase the first character
+
+function ucFirst(str) {
+    str = str[0].toUpperCase() + str.slice(1).toLowerCase();
```
Comment (robhov97) [inline strings.js]: don't change the original param, create a new one

File: strings.js [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+    return str
```
Comment (robhov97) [inline strings.js]: make it one line

File: strings.js [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+let text = 'teXt';
+console.log(ucFirst(text));
+
+// 2. Check for spam
+
+function checkSpam(str) {
+    str = str.toLowerCase();
```
Comment (robhov97) [inline strings.js]: don't change the original param

File: strings.js [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+
+    return str.includes('viagra'.toLowerCase()) || str.includes('XXX'.toLowerCase())
```
Comment (robhov97) [inline strings.js]: 1. Why do you make the 'viagra' lowercase if it's already in lowercase?
2. No need to make lowercase both 'viagra' and 'xxx'. They are your one variable and not coming from outside

**Review by robhov97 — CHANGES_REQUESTED** [review]

File: strings.js (hunk 11 above)
Comment (avagyan98) [reply strings.js]: I need to change 'XXX' to lowercase, because I need to compare that with my string

File: strings.js (hunk 10 above)
Comment (avagyan98) [reply strings.js]: done

File: strings.js (hunk 9 above)
Comment (avagyan98) [reply strings.js]: done

File: strings.js [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+    return str[0].toUpperCase() + str.slice(1).toLowerCase();
+}
+
+let text = 'teXt';
+console.log(ucFirst(text));
+
+// 2. Check for spam
+
+function checkSpam(str) {
+    const LowerCaseStr = str.toLowerCase();
```
Comment (robhov97) [inline strings.js]: use variable names that start with upper case only for classes and constructors

File: strings.js [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+
+    return LowerCaseStr.includes('viagra') || LowerCaseStr.includes('XXX'.toLowerCase())
```
Comment (robhov97) [inline strings.js]: no need to convert 'XXX' to lower case, just write 'xxx'

**Review by robhov97 — CHANGES_REQUESTED** [review]

File: strings.js (hunk 11 above)
Comment (robhov97) [reply strings.js]: no you don't, you already made the input string lowercase

File: strings.js (hunk 8 above)
Comment (avagyan98) [reply strings.js]: done

File: strings.js (hunk 13 above)
Comment (avagyan98) [reply strings.js]: ok)

File: strings.js (hunk 12 above)
Comment (avagyan98) [reply strings.js]: done

File: strings.js (hunk 11 above)
Comment (avagyan98) [reply strings.js]: done

File: strings.js [hunk 14]
```diff
@@ -32,6 +32,24 @@ function extractCurrencyValue(str) {
 }
 
 const money = "$340"
-
 console.log(extractCurrencyValue(money))
 
+
+let words = [];
+let wordStartIndex = 0;
```
Comment (robhov97) [inline strings.js]: move these 2 variables inside of the function, it should be unique for each function call

File: strings.js [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+
+function extractCurrencyValue(str) {
+    return Number(str.slice(1))
+}
+
+const money = "$340"
+console.log(extractCurrencyValue(money))
+
+
+let words = [];
```
Comment (robhov97) [inline strings.js]: Use const for arrays and objects if you don't need to reassign them.
```suggestion
const words = [];
```

**Review by robhov97 — CHANGES_REQUESTED** [review]

File: strings.js (hunk 14 above)
Comment (avagyan98) [reply strings.js]: done)

File: strings.js (hunk 15 above)
Comment (avagyan98) [reply strings.js]: done

**Review by robhov97 — APPROVED** [review]

**Pull request merged**